### PR TITLE
Add Splinter specific PublicKey struct

### PR DESCRIPTION
**Pull request by agunde406**
This struct will be used for internal splinter structs
so we don't need to use the Cylinder's public key as
a part of the splinter API but also is more specific then
a Vec.

### CODE REVIEW

**Review by Caleb-Hill — APPROVED** [review]

File: libsplinter/src/peer/token.rs [hunk 1]
```diff
@@ -65,7 +67,9 @@ impl PeerAuthorizationToken {
     pub fn has_public_key(&self, public_keys: &[Vec<u8>]) -> bool {
         match self {
             PeerAuthorizationToken::Trust { .. } => false,
-            PeerAuthorizationToken::Challenge { public_key } => public_keys.contains(&public_key),
+            PeerAuthorizationToken::Challenge { public_key } => {
+                public_keys.contains(&public_key.as_slice().to_vec())
```
Comment (peterschwarz) [inline libsplinter/src/peer/token.rs:71]: Why does this need to be converted to a vec?  There might be a way to do this without an allocation.

File: libsplinter/src/network/auth/handlers/builder.rs [hunk 2]
```diff
@@ -223,7 +223,9 @@ impl AuthorizationDispatchBuilder {
             let signers_to_use = match &self.local_authorization {
                 Some(ConnectionAuthorizationType::Challenge { public_key }) => {
                     let signer = signers.iter().find(|signer| match signer.public_key() {
-                        Ok(signer_public_key) => signer_public_key.as_slice() == public_key,
+                        Ok(signer_public_key) => {
+                            signer_public_key.as_slice() == public_key.as_slice()
```
Comment (peterschwarz) [inline libsplinter/src/network/auth/handlers/builder.rs:227]: Since `PublicKey` implements `PartialEq`, can we just do the comparison on the public keys?

Comment (agunde406) [reply libsplinter/src/network/auth/handlers/builder.rs:227]: This is comparing splinter::public_key::PublicKey against a cylinder::PublicKey

File: libsplinter/src/peer/token.rs (hunk 1 above)
Comment (agunde406) [reply libsplinter/src/peer/token.rs]: This method is actually not used anywhere so I added a commit to remove it

**Review by peterschwarz — APPROVED** [review]

**Review by isabeltomb — APPROVED** [review]

**Pull request merged**